### ISSUE TITLE
TODOs

### ISSUE THREAD

**Issue by b1ek** [kalaksi/lightkeeper#9]
A list of tasks you can do if you want to help the project.  
Also you can check out the [issues section](https://github.com/kalaksi/lightkeeper/issues) for bug reports!

Dont forget to reference this issue when submitting a task fixing PR!

- [ ] Don't store sensitive info (let modules choose if they contain any of that): `src/cache.rs:62`
- [ ] Open external terminal without freezing the UI thread: `src/command_handler.rs:345,356`
- [ ] Support for multiple commands per module (take a look at `monitor_manager.rs`): `src/command_handler.rs:401,450,483`
- [ ] Check that destination file hasn't changed when uploading remote edits: `src/command_handler.rs:451`
- [ ] Check if `remote_text_editor` and `text_editor` are valid executables: `src/configuration.rs:44,46`
- [ ] Don't block the thread unnecessairly: `src/connection_manager.rs`
- [ ] Validate path string: `src/file_handler.rs:92,99`
- [ ] Refactor `src/frontend/frontend.rs`: `src/frontend/frontend.rs:11`
- [ ] Use HostState instead?: `src/frontend/frontend.rs:35`
- [ ] Use camelCase with QML models: `src/frontend/qt/models/config_manager_model.rs:15`, `src/frontend/qt/models/host_data_manager_model.rs:15`
- [ ] Consider version too: `src/frontend/qt/models/config_manager_model.rs:632`
- [ ] Add parent window id in `src/frontend/qt/models/desktop_portal_model.rs:146,169`
- [ ] Don't use deprecated property: `src/frontend/qt/models/theme_model.rs:48`
- [ ] Use [Kirigami](https://develop.kde.org/docs/getting-started/kirigami/) instead of hex colors?: `src/frontend/qt/models/theme_model.rs:77`

I collected those from code comments that are like this:  
```rs
// TODO: blah blah blah
```

To view all comments like this, do this:
```bash
$ grep -n TODO $(find src -type f)
```

**Comment by kalaksi** [comment]
Just checked one box since I added some path validations.
By the way, any non-trivial code contributions should be discussed first to avoid problems.

**Comment by b1ek** [comment]
> By the way, any non-trivial code contributions should be discussed first to avoid problems.
I agree! I opened this issue just to help collect the TODO stuff (cmon its about 15 thousand lines)

Prime example is the kirrigami thing. Sounds pretty pointless to me and is kind of an overkill
